cumulusci/tasks/github_commit.py

```diff
@@ -0,0 +1,188 @@
+        for root, dirs, files in os.walk(local_dir):
+            for filename in files:
+                if filename.startswith('.'):
+                    # skip hidden files
+                    continue
+                local_file = os.path.join(root, filename)
+                local_file_subpath = local_file[len(local_dir):]
+                if local_file_subpath not in new_tree_target_subpaths:
+                    with io.open(local_file, 'rb') as f:
+                        content = f.read()
+                    if dry_run:
+                        self.logger.info('[dry_run] Skipping creation of ' +
+                            'blob for new file: {}'.format(local_file)
+                        )
+                        blob_sha = None
+                    else:
+                        self.logger.info(
+                            'Creating blob for new file: {}'.format(local_file)
+                        )
+                        blob_sha = self._create_blob(content)
+                        self.logger.debug('Blob created: {}'.format(blob_sha))
+                    new_item = {
+                        'path': '{}/{}'.format(repo_dir, local_file_subpath),
+                        'mode': '100644',
+                        'type': 'blob',
+                        'sha': blob_sha,
+                    }
+                    new_tree_list.append(new_item)
+                    changed = True
+
+        if not changed:
+            self.logger.warning('No changes found, aborting commit')
+            return
+
+        # generate summary of changes
+        self.logger.info('Summary of changes:')
+        new_shas = [item['sha'] for item in new_tree_list]
+        new_paths = [item['path'] for item in new_tree_list]
+        old_paths = [item['path'] for item in tree['tree']]
+        for item in tree['tree']:
+            if item['type'] == 'tree':
+                continue
+            if item['path'] not in new_paths:
+                self.logger.warning('Delete:\t{}'.format(item['path']))
+            elif item['sha'] not in new_shas:
+                self.logger.info('Update:\t{}'.format(item['path']))
+        for item in new_tree_list:
+            if item['path'] not in old_paths:
+                self.logger.info('Add:\t{}'.format(item['path']))
+
+        # create new tree
+        if dry_run:
+            self.logger.info('[dry_run] Skipping creation of new tree')
+        else:
+            self.logger.info('Creating new tree')
+            new_tree = self.repo.create_tree(new_tree_list, None)
+            if not new_tree:
+                raise GithubException('Failed to create tree')
+
+        # create new commit
+        if dry_run:
+            self.logger.info('[dry_run] Skipping creation of new commit')
+        else:
+            self.logger.info('Creating new commit')
+            new_commit = self.repo.create_commit(
+                message='Update Apex docs',
```

Comment:
can we pass this commit message from the calling task, in case we end up wanting to use this utility to commit other things?